src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -1136,5 +1136,24 @@ private void EndTagBlock(Stack<Tuple<HtmlSymbol, SourceLocation>> tags, bool com
             }
             Output(SpanKind.Markup);
         }
+
+        internal static bool IsValidAttributeNameSymbol(HtmlSymbol symbol)
+        {
+            if (symbol == null)
+            {
+                return false;
+            }
+
+            var symbolType = symbol.Type;
+            return symbolType != HtmlSymbolType.WhiteSpace &&
+                symbolType != HtmlSymbolType.NewLine &&
+                symbolType != HtmlSymbolType.CloseAngle &&
+                symbolType != HtmlSymbolType.OpenAngle &&
```

Comment:
Yup, Razor don't like it though :smile:. This is an "almost" spec worthy fix. 

This PR solves all of the use-cases brought forward in issue #137 and takes an extra step in making it spec equivalent. I just filed an issue to cover the open angle derivation: https://github.com/aspnet/Razor/issues/538